sklearn/feature_selection/tests/test_rfe.py

```diff
@@ -612,3 +613,21 @@ def test_multioutput(ClsRFE):
     clf = RandomForestClassifier(n_estimators=5)
     rfe_test = ClsRFE(clf)
     rfe_test.fit(X, y)
+
+
+@pytest.mark.parametrize("ClsRFE", [RFE, RFECV])
+@pytest.mark.parametrize("PLSEstimator", [CCA, PLSCanonical, PLSRegression])
+def test_rfe_pls(ClsRFE, PLSEstimator):
+    """Check the behaviour of RFE with PLS estimators.
+
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/12410
+    """
+    X, y = make_friedman1(n_samples=50, n_features=10, random_state=0)
+    estimator = PLSEstimator(n_components=1)
+    selector = ClsRFE(estimator, step=1)
+    with pytest.warns(None) as records:
+        selector.fit(X, y)
+    # we should not warn by accessing `coef_` of the PLS estimators
+    assert len(records) == 0
```

Comment:
```suggestion
    # We must not warn by accessing `coef_` attribute of PLSEstimator
    with pytest.warns(None) as records:
        # fit calls access `coef_` under the hood
        selector.fit(X, y)
    assert len(records) == 0
```

Reply:
I dropped the inner comment because it will actually check for `_coef_`